### PR TITLE
Upgrade to Helm v3.13.3

### PR DESCRIPTION
**Pull request by lackstein**
Upgrade the Helm Operator from Helm v3.6.3 to v3.13.3. Newer versions of Helm include updates to functions exposed in go templates. This should resolve an issue causing Flux to fail to deploy the `ingress-nginx` chart v4.9.0.

See https://github.com/kubernetes/ingress-nginx/pull/10659#issuecomment-1854851888 and [FDN-101](https://fossa.atlassian.net/browse/FDN-101).

[FDN-101]: https://fossa.atlassian.net/browse/FDN-101?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -30,6 +33,140 @@ require (
 	sigs.k8s.io/yaml v1.2.0
 )
 
+require (
```
Comment (elldritch) [inline go.mod:33]: [nit] Should this be part of the same `require` block as above?

File: cmd/helm-operator/main.go [hunk 2]
```diff
@@ -28,10 +28,10 @@ import (
 	helmv3 "github.com/fluxcd/helm-operator/pkg/helm/v3"
 	v3 "github.com/fluxcd/helm-operator/pkg/helm/v3"
 	daemonhttp "github.com/fluxcd/helm-operator/pkg/http/daemon"
-	"github.com/fluxcd/helm-operator/pkg/operator"
 	"github.com/fluxcd/helm-operator/pkg/release"
 	"github.com/fluxcd/helm-operator/pkg/status"
 	"github.com/fluxcd/helm-operator/pkg/utils"
+	"github.com/fossas/helm-operator/pkg/operator"
```
Comment (elldritch) [inline cmd/helm-operator/main.go]: Here and elsewhere, why are we now using the forked version of `pkg/operator`? I don't see any material changes to that package.

**Review by elldritch — COMMENTED** [review]
I don't think I fully understand this PR - it's a bit confusing to tease apart which bits are cleanup and which bits are intentional logic changes.

I think the version upgrades are the meat of the changes here, right?

**Comment by lackstein** [comment]
@elldritch yes, the change to the Helm version in `go.mod` is the important bit.

File: cmd/helm-operator/main.go (hunk 2 above)
Comment (lackstein) [reply cmd/helm-operator/main.go]: They import an internal locking package and I can't for the life of me get Go to build without changing this import. I've tried multiple incantations of replace directives and have had no luck. 

Error would be some variant of
```
package github.com/fossas/helm-operator/internal/lockedfile
        internal/lockedfile/lockedfile_filelock.go:12:2: use of internal package github.com/fluxcd/helm-operator/internal/lockedfile/internal/filelock not allowed
package github.com/fossas/helm-operator/pkg/operator
        pkg/operator/operator.go:24:2: use of internal package github.com/fluxcd/helm-operator/internal/lockedfile not allowed
```
or
```
package github.com/fossas/helm-operator/pkg/operator
        imports github.com/fluxcd/helm-operator/internal/lockedfile
        vendor/github.com/fluxcd/helm-operator/internal/lockedfile/lockedfile_filelock.go:13:2: use of internal package github.com/fossas/helm-operator/internal/lockedfile/internal/filelock not allowed
```

File: go.mod (hunk 1 above)
Comment (lackstein) [reply go.mod:33]: This was autogenerated by `go mod tidy`. I don't write enough Golang to know what would be idiomatic.

**Comment by elldritch** [comment]
@lackstein I've pushed another commit on this branch that minimizes the diff. This builds for me locally - 🤞🏼 that it works in CI.

**Comment by elldritch** [comment]
Oh, I see.

**Review by elldritch — APPROVED** [review]
There we go. How's this, @lackstein?

**Pull request merged**